### PR TITLE
update monsterlists to v0.24

### PR DESCRIPTION
**Pull request by jmbjr**
not sure if you maintain these lists anymore, but I managed to update the  monsters.py script to crawl v0.24.

I think there's a couple issues with this.  
1. I updated to python3, which may not be desired
2. I hard-coded the path to my crawl repo, which again is probably not desired.

both of these are in the first commit to convert to python3 so probably easy to remove.

For version 0.18, I modified the HP output tables to be more in-line with the big update away from HD being part of the HP equation. I didn't do a ton of verification for this and relied on the commit explanation: https://github.com/crawl/crawl/commit/4878628b8a4504d368c537c21ed9ca6e0c089aaa

I didn't handle the other scripts at this time.

### CODE REVIEW

File: monsters.py [hunk 1]
```diff
@@ -405,7 +405,7 @@ def main():
     for m in olddata:
         mons = Monster(*m)
         old_monsters[mons.id] = mons
-    monsters = sorted(all_monsters.itervalues(), key=attrgetter('id'))
+    monsters = sorted(iter(list(all_monsters.values())), key=attrgetter('id'))
```
Comment (ChrisOelmueller) [inline monsters.py:410]: This looks like simply `sorted(all_monsters.values(), key=attrgetter('id'))` should work?

File: monsters.py [hunk 2]
```diff
@@ -514,7 +514,7 @@ def print_everything():
         'speed': print_speed,
     }
 
-    for (abbr, fn) in fnmap.iteritems():
+    for (abbr, fn) in list(fnmap.items()):
```
Comment (ChrisOelmueller) [inline monsters.py:520]: Most of the `items()` changes in this commit don't actually need to be wrapped in a `list()` call. At least from what I can see and remember, order is either:
- not guaranteed anyways
- not important

When simply iterating over stuff, it's not required to know the entire data structure beforehand :)

File: monsters.py [hunk 3]
```diff
@@ -224,29 +224,27 @@ def __repr__(self):
...
 
     @property
     def fixed(self):
         return self.min == self.max
 
     def __repr__(self):
-        return "(%3s..[%3s]..%3s)   %2dd%-2d[%+d]%+4d" % (
-                self.min, self.avg, self.max,
-                self.hd, self.min_hp, self.rand_hp, self.add_hp)
+        return "(%3s..[%3s]..%3s)   (%3s..[%3s]..%3s)" % (
```
Comment (ChrisOelmueller) [inline monsters.py:245]: Eek, that old format-string :+1:

**Comment by ChrisOelmueller** [comment]
Thanks for putting work in here! I was not actively maintaining this, and actually used Monsters more as pet project for whenever I needed quick "global" information. If somebody pops up and finds something useful inside, I try to get things up to speed occasionally :)

In this case, you're correct regarding the hard-coded file paths. That's not ideal and I might toss in some basic argparse (or better guessing) to replace it with. Python3 conversion is fine, I added a few hints regarding that in the commit.

**Review by ChrisOelmueller — APPROVED** [review]

**Comment by ChrisOelmueller** [comment]
I've modified the approach to some of the issues you resolved and pushed those changes plus your updates to `master`. Until I find some more time, you may have to keep around a local modification for finding the correct mon-data.h path on your machine.

I'm going to close here, thanks again for the PR!

**Comment by jmbjr** [comment]
thanks for the quick turnaround and fixing some of my kludges!  I really like these lists and kept saying that I'd take a stab at trying to get it updated for a while now.  The HP update to v0.18 was interesting since I didn't know till looking at the code that uniques had variable HP now.

thanks again
john(stein)

**Pull request closed without merging**